### PR TITLE
Untag dispatch vector<👻>::_Insert

### PR DESCRIPTION
**Pull request by AlexGuteniev**
Towards #189

### CODE REVIEW

File: stl/inc/vector [hunk 1]
```diff
@@ -3082,26 +3082,23 @@ public:
 
     template <class _Iter, enable_if_t<_Is_iterator_v<_Iter>, int> = 0>
     _CONSTEXPR20 iterator insert(const_iterator _Where, _Iter _First, _Iter _Last) {
-        difference_type _Off = _Where - begin();
-        _Insert(_Where, _First, _Last, _Iter_cat_t<_Iter>{});
-        return begin() + _Off;
-    }
+        const difference_type _Off_ret = _Where - begin();
```
Comment (miscco) [inline stl/inc/vector:3085]: I would prefer if we could keep this as `_Off` and give the temporary in the fwd iterator branch a new name

Comment (miscco) [reply stl/inc/vector:3085]: Srew that, I missed that we did decrement it below :(

**Review by miscco — APPROVED** [review]

File: stl/inc/vector (hunk 1 above)
Comment (CaseyCarter) [inline stl/inc/vector:3085]: I'm not fond of the name `_Off_ret` ("returned offset"). `_Saved_offset` would be more consistent with our usual naming of locals. 
```suggestion
        const difference_type _Saved_offset = _Where - begin();
```

File: stl/inc/vector [hunk 2]
```diff
@@ -3082,26 +3082,23 @@ public:
...
-    _CONSTEXPR20 void _Insert(const_iterator _Where, _Iter _First, _Iter _Last, input_iterator_tag) {
-        difference_type _Off = _Where - begin();
+        if constexpr (_Is_fwd_iter_v<_Iter>) {
+            _Adl_verify_range(_First, _Last);
+            const auto _Count    = _Convert_size<size_type>(static_cast<size_t>(_STD distance(_First, _Last)));
+            const size_type _Off = _Insert_x(_Where, _Count);
+            _Copy_unchecked(
+                _Get_unwrapped(_First), _Get_unwrapped(_Last), begin() + static_cast<difference_type>(_Off));
+        } else {
+            difference_type _Off = _Off_ret;
```
Comment (CaseyCarter) [inline stl/inc/vector:3094]: Ditto:
```suggestion
            difference_type _Off = _Saved_offset;
```

File: stl/inc/vector [hunk 3]
```diff
@@ -3082,26 +3082,23 @@ public:
...
         }
-    }
 
-    template <class _Iter>
-    _CONSTEXPR20 void _Insert(const_iterator _Where, _Iter _First, _Iter _Last, forward_iterator_tag) {
-        _Adl_verify_range(_First, _Last);
-        auto _Count    = _Convert_size<size_type>(static_cast<size_t>(_STD distance(_First, _Last)));
-        size_type _Off = _Insert_x(_Where, _Count);
-        _Copy_unchecked(_Get_unwrapped(_First), _Get_unwrapped(_Last), begin() + static_cast<difference_type>(_Off));
+        return begin() + _Off_ret;
```
Comment (CaseyCarter) [inline stl/inc/vector:3101]: Ditto:
```suggestion
        return begin() + _Saved_offset;
```

**Review by CaseyCarter — APPROVED** [review]
I'll apply my renaming suggestion and approve.

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Comment by StephanTLavavej** [comment]
:ghost: :smile_cat: :ghost:
===

**Pull request merged**